test/fs/test_close.c

```diff
@@ -0,0 +1,33 @@
+#include <emscripten/emscripten.h>
+#include <stdio.h>
+
+int main() {
+
+    EM_ASM(
+        FS.writeFile("testfile", 'a=1\nb=2\n');
+        FS.mkdir("/testdir");
+        var file = FS.open("testfile", "r");
+        var error = FS.close(file);
+        assert(!error);
+
+        file = FS.open("/testdir", "r");
+        error = FS.close(file);
+        assert(!error);
+
+#if WASMFS
+        error = FS.close(file);
+        console.log("Error: ", error);
+        assert(error === 8 /* EBADF */);
```

Comment:
Did you decide that we want the WASMFS version of these API to return a code rather than throwing?   Why not just stick with throwing for compatability purposes?

Reply:
I don't think we had a firm plan here, but IIRC it seemed to @tlively , @jameshu15869 and myself that we don't want to do things like return inspectable JS objects from the JS FS API. That would be needed for full compatibility (the old API returns Node objects with lots of properties), but we hope we can live without it. Exceptions is related, since I was thinking we'd just return an error code, like the POSIX APIs do, and not JS error objects. But you're right that that might have compatibility issues, so maybe it is worth reconsidering now, or later.

Reply:
I think its probably worth going for compatibility first and then figuring out what is we can improve later (maybe leave some TODOs in the code to make it clear when something exists just for compat).

Reply:
Hmm, yeah, I think that's probably best. Returning integers instead of objects is ok because the main use case is `FS.write(FS.open(..))` etc. where the user never inspects the return from `open`. But replacing a throw with an integer is much more surprising. So let's keep throwing here.